src/test/java/com/bobocode/hoverla/bring/context/BeanValidatorTest.java

```diff
@@ -0,0 +1,160 @@
+import java.util.List;
+import java.util.Map;
+
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class BeanValidatorTest {
+
+    private static final String BD1 = "beanDef1";
+    private static final String BD2 = "beanDef2";
+    private static final String BD3 = "beanDef3";
+    private static final String BD4 = "beanDef4";
+    private List<BeanDefinition> beanDefinitionList;
+    private BeanValidator beanValidator;
+
+    @BeforeEach
+    public void setUp() {
+        beanValidator = new BeanValidator();
+        BeanDefinition beanDef4 = prepareDefinition(BD4, Integer.class);
+        BeanDefinition beanDef3 = prepareDefinition(BD3, Byte.class, beanDef4);
+        BeanDefinition beanDef2 = prepareDefinition(BD2, Short.class, beanDef3, beanDef4);
+        BeanDefinition beanDef1 = prepareDefinition(BD1, String.class, beanDef2, beanDef3);
+        beanDefinitionList = new ArrayList<>(Arrays.asList(beanDef1, beanDef2, beanDef3, beanDef4));
+
+    }
+
+    @Test
+    @DisplayName("All validations passed success")
+    void validationWithoutErrors() {
+        beanValidator.validate(beanDefinitionList);
+    }
+
+    @Test
+    @DisplayName("Fails when bean definition list is null ")
+    void validationWhenListIsNull() {
+        String expectedMessage = "Bean definition list is null";
+
+        var ex = Assertions.assertThrows(NullPointerException.class,
+                () -> beanValidator.validate(null));
+        Assertions.assertTrue(ex.getMessage().contains(expectedMessage));
+    }
+
+    @ParameterizedTest
+    @NullAndEmptySource
+    @ValueSource(strings = {" ", "a a"})
+    @DisplayName("Fails on name validation, it's null")
+    void beanNameIsNull(String name) {
+        String expectedMessage = "Bean name for String class is null, empty or contains spaces";
+        BeanDefinition beanDefinition = prepareDefinition(name, String.class);
+        beanDefinitionList.add(beanDefinition);
+
+        assertExceptionAndMessage(expectedMessage);
+    }
+
+    @Test
+    @DisplayName("Fails on dependency validation when found more than 1 dependency with the same name")
+    void duplicateNamesInDependencies() {
+        String expectedMessage = "Context contains beans with duplicate names:";
+        BeanDefinition beanDefinition = prepareDefinition(BD1, Long.class);
+        BeanDefinition beanDefinitionWithConflictDependencies = prepareDefinition("BD5", String.class,
+                beanDefinition);
+        beanDefinitionList.add(beanDefinition);
+        beanDefinitionList.add(beanDefinitionWithConflictDependencies);
+
+        assertExceptionAndMessage(expectedMessage);
+    }
+
+    @Test
+    @DisplayName("Fails on dependency validation when the found dependency type is not suitable for the required type")
+    void unsuitableTypeForDependency() {
+        String expectedMessage = "Types are not coincide for required type";
```

Comment:
(nit) I'd change work coincide to something more common :)

Reply:
Done